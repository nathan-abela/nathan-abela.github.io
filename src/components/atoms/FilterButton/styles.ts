import styled from 'styled-components';

interface StyledFilterButtonProps {
	isActive: boolean;
}

const StyledFilterButton = styled.button<StyledFilterButtonProps>`
	padding: 8px 16px;
	margin: 0 8px 8px 0;
	border: 1px solid ${({ isActive, theme }) => (isActive ? theme.primary : theme.text)};
	border-radius: 4px;
	background-color: ${({ isActive, theme }) => (isActive ? theme.primary : theme.body)};
	color: ${({ isActive, theme }) => (isActive ? theme.body : theme.text)};
	cursor: pointer;
	transition: background-color 0.3s ease, color 0.3s ease;

	&:hover {
		background-color: ${({ theme }) => `${theme.primary}F2`};
		color: ${({ theme }) => theme.headline};
	}

	@media screen and (max-width: 425px) {
		font-size: 14px;
	}
`;

export { StyledFilterButton };
