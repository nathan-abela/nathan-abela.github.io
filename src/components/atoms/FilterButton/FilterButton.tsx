import { FC } from 'react';

import { StyledFilterButton } from './styles';

interface FilterButtonProps {
	onClick: () => void;
	isActive: boolean;
	children: React.ReactNode;
}

const FilterButton: FC<FilterButtonProps> = ({ onClick, isActive, children }) => {
	return (
		<StyledFilterButton onClick={onClick} isActive={isActive}>
			{children}
		</StyledFilterButton>
	);
};

export { FilterButton };
