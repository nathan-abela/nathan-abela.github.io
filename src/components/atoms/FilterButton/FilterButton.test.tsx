import { render } from '@testing-library/react';

import { FilterButton } from './FilterButton';

describe('Filter Button', () => {
	it('renders filter button with isActive is true', () => {
		const { container } = render(
			<FilterButton onClick={jest.fn} isActive={true}>
				Filter button text
			</FilterButton>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders filter button with isActive is false', () => {
		const { container } = render(
			<FilterButton onClick={jest.fn} isActive={false}>
				Filter button text
			</FilterButton>
		);

		expect(container).toMatchSnapshot();
	});
});
