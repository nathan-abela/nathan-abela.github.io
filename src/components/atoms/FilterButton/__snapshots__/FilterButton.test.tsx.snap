// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Button renders correctly when isActive is true 1`] = `
<div>
  <button
    class="styles__StyledFilterButton-sc-nr6hfb-0 hPZrEs"
  >
    Filter
  </button>
</div>
`;

exports[`Filter Button renders filter button with isActive is false 1`] = `
<div>
  <button
    class="styles__StyledFilterButton-sc-nr6hfb-0 hPZrEs"
  >
    Filter button text
  </button>
</div>
`;

exports[`Filter Button renders filter button with isActive is true 1`] = `
<div>
  <button
    class="styles__StyledFilterButton-sc-nr6hfb-0 hPZrEs"
  >
    Filter button text
  </button>
</div>
`;
